cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -426,9 +439,20 @@ public Set<ProgramId> filterProgramsExistence(Collection<ProgramId> programIds)
   }
 
 
-  public void writeApplication(String namespaceId, String appId, String versionId, ApplicationSpecification spec)
+  public void writeApplication(String namespaceId, String appId, String versionId, ApplicationSpecification spec,
+                                Long created, @Nullable String author, @Nullable String changeSummary)
+    throws IOException {
+    writeApplicationSerialized(namespaceId, appId, versionId, GSON.toJson(new ApplicationMeta(appId, spec,
+                                                                                              changeSummary, created,
+                                                                                              author)), created,
+                               author, changeSummary);
+  }
+
+  public void updateLatestApplication(String namespaceId, String appId, String versionId)
```

Comment:
I only see this being called from the addApplication() method, is it being called anywhere else? If not, I think it's more clear to put it directly in addApplication() rather than having this method, as we only want to set 'latest' to false in very specific situations.